src/mono/wasm/runtime/driver.c

```diff
@@ -685,23 +685,27 @@ mono_wasm_invoke_method_ref (MonoMethod *method, MonoObject **this_arg_in, void
 	MONO_EXIT_GC_UNSAFE;
 }
 
-EMSCRIPTEN_KEEPALIVE MonoObject*
-mono_wasm_invoke_method_bound (MonoMethod *method, void* args)// JSMarshalerArguments
+EMSCRIPTEN_KEEPALIVE int
+mono_wasm_invoke_method_bound (MonoMethod *method, void* args /*JSMarshalerArguments*/, MonoObject **_out_exc)
 {
-	MonoObject *exc = NULL;
-	MonoObject *res;
-
+	PPVOLATILE(MonoObject) out_exc = _out_exc;
 	void *invoke_args[1] = { args };
+	int is_err = 0;
+
+	MONO_ENTER_GC_UNSAFE;
+	mono_runtime_invoke (method, NULL, invoke_args, (MonoObject **)out_exc);
```

Comment:
I don't understand why this pointer makes a round-trip through out_exc, it seems unnecessary